### PR TITLE
Add SSO button

### PR DESCRIPTION
**Pull request by mateoclarke**
Closes https://github.com/cityofaustin/atd-data-tech/issues/4420
Closes https://github.com/cityofaustin/atd-data-tech/issues/4419

Dec-08-2020 01-37-27

### CODE REVIEW

File: moped-editor/src/index.js [hunk 1]
```diff
@@ -4,6 +4,48 @@ import { BrowserRouter } from "react-router-dom";
...
+      scope: [
+        "aws.cognito.signin.user.admin",
+        "email",
+        "openid",
+        "phone",
+        "profile",
+      ],
+      redirectSignIn: "https://localhost:3000/moped/session/signin",
+      redirectSignOut: "https://localhost:3000/moped/session/signin",
+      responseType: "code",
```
Comment (sergiogcx) [inline moped-editor/src/index.js:36]: Those URLs should not be hard-coded, we need to be able to pass those as variables (if they are not already)

File: moped-editor/src/auth/useAuthentication.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    return () => {
+      Hub.remove("auth", handler);
+    };
+  }, [refreshState]);
+
+  const signIn = useCallback(() => {
+    Auth.federatedSignIn({ provider: "COGNITO" }).catch(err => {
+      setError(err);
+    });
+  }, []);
```
Comment (sergiogcx) [inline moped-editor/src/auth/useAuthentication.js:91]: Very very impressive work here! 🙌 🥇

**Review by sergiogcx — APPROVED** [review]
You made it look easy! 😎 Thank you for working on this!

File: moped-editor/src/index.js (hunk 1 above)
Comment (mateoclarke) [reply moped-editor/src/index.js:36]: yeah, thats the next PR

File: moped-editor/src/auth/user.js [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import React from "react";
-import { Auth } from "aws-amplify";
+import { Auth, Hub } from "aws-amplify";
```
Comment (mddilley) [inline moped-editor/src/auth/user.js:2]: This `Hub` import creates one console warning that it is never used.

**Review by mddilley — APPROVED** [review]
Awesome stuff here, Mateo! Lots here to learn from. 👍  I logged in with my CoA credentials and saw my AzureAD username in the staging user pool just as expected. 🙌  
Screen Shot 2020-12-08 at 11 52 42 AM

One thing that I ran into while testing is wondering how SSO users will interact with the user management features. I'll create an issue for how to handle password updates (maybe just not showing the UI if logged in with AD? Or handle in the API?). 

Also, I noticed that the `email_verified` attribute is `false` for AD users which restricts them from utilizing the user API. Not familiar enough with how the SSO workflow handles that attribute so either you or @sergiogcx might have a good suggestion. https://github.com/cityofaustin/atd-moped/blob/3131d2e12a9a6495106d7d1af2c330a1b055a475/moped-api/claims.py#L97

**Comment by mateoclarke** [comment]
@amenity, I forgot to add you to review. My apologies. The features in the PR also won't work in staging yet (so you can't test it), but working on that now.

**Pull request merged**